frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1878,6 +1890,128 @@ function ReaderFooter:genAlignmentMenuItems(value)
     }
 end
 
+function ReaderFooter:getNamedPresetMenuItems()
+    local footer_presets = G_reader_settings:readSetting("footer_presets", {})
+    local items = {
+        {
+            text = _("Create new preset from current settings"),
+            keep_menu_open = true,
+            callback = function(touchmenu_instance)
+                self:createPresetFromCurrentSettings(touchmenu_instance)
+            end,
+            separator = true,
+        },
+    }
+    -- Add menu items for each preset
+    for preset_name in ffiUtil.orderedPairs(footer_presets) do
+        table.insert(items, {
+            text = preset_name,
+            keep_menu_open = true,
+            callback = function()
+                self:loadFromNamedPreset(preset_name)
+            end,
+            hold_callback = function(touchmenu_instance)
+                UIManager:show(MultiConfirmBox:new{
+                    text = T(_("What would you like to do with preset '%1'?"), preset_name),
+                    choice1_text = _("Delete"),
+                    choice1_callback = function()
+                        self:deleteNamedPreset(preset_name)
```

Comment:
No need for a single-line method, can be done right here:
`footer_presets[preset_name] = nil`